modules/stochastic_tools/src/controls/MultiAppCommandLineControl.C

```diff
@@ -106,14 +106,29 @@ MultiAppCommandLineControl::execute()
                  _param_names.size(),
                  ").");
 
-    for (unsigned int row = 0; row < matrix.m(); ++row)
+    if (std::dynamic_pointer_cast<SamplerFullSolveMultiApp>(_multi_app) == nullptr)
```

Comment:
Can you explain what you are trying to accomplish?

Reply:
The command line option is like name_a = 1, name_b = 2, name_c = 3..... The name is the same for each sampler and only the values are different. If we store them all as strings, the mem usage is huge. So I think I just need to store only names into the cli_arg and them in the ``getCommandLineArgsParamHelper`` I added the name to values together. ``getCommandLineArgsParamHelper`` is a virtual function defined in ``MultiApp`, so I do no want to change this for standard multiApp but just for ``SamplerFullSolveMultiApp``.

Reply:
OK, can you write some detailed comments above this if statement and in the SamplerFullSolveMultiApp::getCommandLineArgsParamHelper so we (I) can remember why this is done.

Reply:
Sure, I will do it. 

Do you have any other comments that I need to address?

Reply:
No, I am good with what you did.